<?xml version="1.0" ?>
<launch>
  <arg name="robot_ip"               default="192.168.3.102"/>
  <arg name="load_gripper"           default="true" />
  <arg name="use_interactive_marker" default="false" />
  <arg name="use_gripper_gui"        default="true" />
  <arg name="load_franka_control"    default="true" />

  <!-- Bringup franka_interactive_bringup.laucnh -->
  <group if="$(arg load_franka_control)">
    <include file="$(find franka_interactive_controllers)/launch/franka_interactive_bringup.launch" >
      <arg name="robot_ip" value="$(arg robot_ip)" />
      <arg name="load_gripper" value="$(arg load_gripper)" />
      <arg name="use_gripper_gui" value="$(arg use_gripper_gui)" />
      <arg name="bringup_rviz" value="true" />
    </include>
  </group>
  
  <!-- Load desired controller-->  
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="cartesian_pose_impedance_controller"/>
  <rosparam  ns="cartesian_pose_impedance_controller" command="load" file="$(find franka_interactive_controllers)/config/pose_impedance_control_additional_params.yaml"/>

  <!-- IF interactive marker= true: Use interactive marker to define desired pose for impedance (stiffness + damping compensation) controller --> 
  <group if="$(arg use_interactive_marker)">
    <node name="interactive_marker" pkg="franka_interactive_controllers" type="interactive_marker.py" required="true" output="screen">
      <param name="link_name" value="panda_link0" />
      <remap from="desired_pose" to="/cartesian_impedance_controller/desired_pose" />
    </node>
  </group>

</launch>
